### ISSUE TITLE
ft_test not specific when run with dependency

### ISSUE THREAD

**Issue by britta-wstnr** [fieldtrip/fieldtrip#1764]
With @schoffelen, we realized today that the `ft_test` routines are not strictly specific when run with the `dependency` flag: for example specifying `dependency ft_inv` triggers tests for all `ft_inv*`, so also for `ft_inverse_lcmv`, `ft_inverse_dics` etc.
 
Not sure if that should/could be changed, but we decided it should be documented.

**Comment by robertoostenveld** [comment]
good point. I suspect that the code now does something dumb that resembles a plain `grep` (i.e. a substring search), perhaps that could be changed into a better `regexp`.

**Comment by robertoostenveld** [comment]
the code responsible for this is at 
https://github.com/fieldtrip/fieldtrip/blob/ab4f1ac9d8bfe202b8ddbd2dc0618ede8f07e254/utilities/private/ft_test_run.m#L114
